paper-server/src/main/java/io/papermc/paper/world/biome/PaperBiomeMobSpawning.java

```diff
@@ -0,0 +1,54 @@
+package io.papermc.paper.world.biome;
+
+import com.google.common.collect.ImmutableMap;
+import io.papermc.paper.util.MCUtil;
+import net.minecraft.world.level.biome.Biome;
+import net.minecraft.world.level.biome.MobSpawnSettings;
+import org.bukkit.craftbukkit.entity.CraftEntityType;
+import org.bukkit.craftbukkit.util.CraftSpawnCategory;
+import org.bukkit.entity.EntityType;
+import org.bukkit.entity.SpawnCategory;
+import java.util.List;
+import java.util.Map;
+
+public class PaperBiomeMobSpawning implements BiomeMobSpawning {
+    private final MobSpawnSettings settings;
+    private final ImmutableMap<SpawnCategory, List<MobData>> spawners;
```

Comment:
What is the rational between building the spawners map in the ctor and the costs in the getter?

Reply:
SpawnCategory is an enum on both nms and bukkit. Entity type is a registry value, which feels really wrong to cache

Reply:
Fair enough :+1: